File: bundles/at.bestsolution.tasktracker.dto/src/at/bestsolution/tasktracker/emapping/webservice/dto/DTOChangeSet.java
package at.bestsolution.tasktracker.emapping.webservice.dto;

public class DTOChangeSet {
	/*
	 * Meta data
	 */
	private String metaClassname = "ChangeSet";
	private boolean metaProxy;

	public void setMetaProxy(boolean metaProxy) {
		this.metaProxy = metaProxy;
	}

	public boolean isMetaProxy() {
		return metaProxy;
	}

	public void setMetaClassname(String metaClassname) {
		this.metaClassname = metaClassname;
	}

	public String getMetaClassname() {
		return metaClassname;
	}

	public static DTOChangeSet newProxy(long id) {
		DTOChangeSet dto = new DTOChangeSet();
		dto.setSid(id);
		dto.setMetaProxy(true);
		return dto;
	}

	/*
	 * Attributes
	 */
	private long sid;
	private java.lang.String id;
	private java.lang.String comment;

	/*
	 * References
	 */
	private DTOTask task;
	private DTOSourceRepository sourceRepository;
	private DTOPerson person;

	public long getSid() {
		return this.sid;
	}

	public void setSid(long sid) {
		this.sid = sid;
	}
	public java.lang.String getId() {
		return this.id;
	}

	public void setId(java.lang.String id) {
		this.id = id;
	}
	public java.lang.String getComment() {
		return this.comment;
	}

	public void setComment(java.lang.String comment) {
		this.comment = comment;
	}

	public DTOTask getTask() {
		return this.task;
	}

	public void setTask(DTOTask task) {
		this.task = task;
	}

	public DTOSourceRepository getSourceRepository() {
		return this.sourceRepository;
	}

	public void setSourceRepository(DTOSourceRepository sourceRepository) {
		this.sourceRepository = sourceRepository;
	}

	public DTOPerson getPerson() {
		return this.person;
	}

	public void setPerson(DTOPerson person) {
		this.person = person;
	}

}

File: bundles/at.bestsolution.tasktracker.server/static/ts/model/DTOChangeSet.ts
/// <reference path="DTOTask.ts"/>
/// <reference path="DTOSourceRepository.ts"/>
/// <reference path="DTOPerson.ts"/>

class DTOChangeSet {
	metaClassname : String = "ChangeSet"
	metaProxy : boolean = false

	// simple attributes
	sid : number
	id : string
	comment : string

	// reference attributes
	task : DTOTask
	sourceRepository : DTOSourceRepository
	person : DTOPerson
	constructor(jsonObject? : any) {
		if(jsonObject) {
			this.metaProxy = jsonObject.metaProxy

			this.sid = jsonObject.sid;
			this.id = jsonObject.id;
			this.comment = jsonObject.comment;

			this.task = jsonObject.task ? new DTOTask(jsonObject.task) : null;
			this.sourceRepository = jsonObject.sourceRepository ? new DTOSourceRepository(jsonObject.sourceRepository) : null;
			this.person = jsonObject.person ? new DTOPerson(jsonObject.person) : null;
		}
	}
}
